### PR TITLE
Add core/fallback encoder behaviour

### PR DESCRIPTION
**Pull request by waffle87**
## Description

Title. Am open to suggestions about checking for `_user()` function.

## Types of Changes

- [x] Core

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
yeah, for whatever reason the unit tests do NOT like tap_code or tap_code_delay.

File: quantum/encoder.c [hunk 1]
```diff
@@ -77,7 +77,23 @@ __attribute__((weak)) bool encoder_update_user(uint8_t index, bool clockwise) {
...
 __attribute__((weak)) bool encoder_update_kb(uint8_t index, bool clockwise) {
-    return encoder_update_user(index, clockwise);
+    bool res = encoder_update_user(index, clockwise);
+    if (res) {
+        if (clockwise) {
+#ifdef EXTRAKEY_ENABLE
+            tap_code_delay(KC_VOLU, 10);
+#else
+            tap_code_delay(KC_PGDN, 10);
+#endif
```
Comment (tzarc) [inline quantum/encoder.c:87]:
```suggestion
#if defined(EXTRAKEY_ENABLE)
            tap_code_delay(KC_VOLU, 10);
#elif defined(MOUSEKEY_ENABLE)
            tap_code_delay(KC_MS_WH_UP, 10);
#else
            tap_code_delay(KC_PGDN, 10);
#endif
```

File: quantum/encoder.c [hunk 2]
```diff
@@ -77,7 +77,23 @@ __attribute__((weak)) bool encoder_update_user(uint8_t index, bool clockwise) {
...
+        } else {
+#ifdef EXTRAKEY_ENABLE
+            tap_code_delay(KC_VOLD, 10);
+#else
+            tap_code_delay(KC_PGUP, 10);
+#endif
```
Comment (tzarc) [inline quantum/encoder.c:93]:
```suggestion
#if defined(EXTRAKEY_ENABLE)
            tap_code_delay(KC_VOLD, 10);
#elif defined(MOUSEKEY_ENABLE)
            tap_code_delay(KC_MS_WH_DOWN, 10);
#else
            tap_code_delay(KC_PGUP, 10);
#endif
```

**Review by tzarc — COMMENTED** [review]
What about a slightly more graceful fallback?

**Comment by drashna** [comment]
wrapping the block in `#ifndef ENCODER_TESTS` works to prevent the compilation issue for the tests, at least. Not great, but works.

File: quantum/encoder.c (hunk 1 above)
Comment (drashna) [reply quantum/encoder.c:87]: could probably also use `ENCODER_MAP_KEY_DELAY` instead of 10. but that gets set as 0 by default.

**Comment by waffle87** [comment]
Yeah, bit of conditional spaghetti, but probably better than failing for no reason for anyone in the future ha.

**Comment by drashna** [comment]
Ideally, I'd love to sort out why.  But above my pay grade for understanding. 

Also, it's why you beat me to opening the PR.  (`make test:encoder` will usually run locally, for instance, macOS being the exception).

**Review by drashna — APPROVED** [review]

**Pull request merged**